### PR TITLE
Reduce memory footprint of integration test by not starting unneeded containers

### PR DESCRIPTION
**Pull request by maytasm**
Reduce memory footprint of integration test by not starting unneeded containers

### Description

Additional Druid router services with permissive tls, no client auth tls, and custom check tls does not need to be started when test group is not "security". Hence, those should only be started when for running integration test group security.

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: integration-tests/README.md [hunk 1]
```diff
@@ -74,17 +74,21 @@ Docker compose yamls located in "docker" folder
 
 docker-compose.base.yml - Base file that defines all containers for integration test
 
-docker-compose.yml - Defines minimal Druid cluster that can be used for non cluster tests
+docker-compose.yml - Defines Druid cluster with default configuration that is use for running integration tests in Travis CI.
```
Comment (clintropolis) [inline integration-tests/README.md:77]: nit: '... that is *used* ...'

File: integration-tests/README.md [hunk 2]
```diff
@@ -74,17 +74,21 @@ Docker compose yamls located in "docker" folder
...
 
-docker-compose.override-env.yml - the same configuration as docker-compose.yml + override-env variable that needed to run cloud tests 
+docker-compose.override-env.yml - Defines Druid cluster with default configuration plus any additional and/or overriden configurations from override-env file.
 
     // OVERRIDE_ENV - variable that must contains path to Druid configuration file 
     OVERRIDE_ENV=./environment-configs/override-examples/s3 docker-compose -f docker-compose.override-env.yml up
     
+docker-compose.security.yml - Defines three additional Druid router services with permissive tls, no client auth tls, and custom check tls respectively. This is only needed for the "security" group integration test. 
+
+    docker-compose -f docker-compose.druid-hadoop.yml up
```
Comment (clintropolis) [inline integration-tests/README.md:90]: nit: wrong command?

File: integration-tests/README.md (hunk 1 above)
Comment (maytasm) [reply integration-tests/README.md:77]: Done

File: integration-tests/README.md (hunk 2 above)
Comment (maytasm) [reply integration-tests/README.md:90]: oops. Fixed

**Comment by maytasm** [comment]
@clintropolis Addressed your comments and some more minor change in the README.

**Pull request merged**